src/include/processor/operator/persistent/parquet_file_writer.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include "arrow/io/file.h"
+#include "processor/operator/persistent/csv_parquet_writer.h"
+#include "processor/operator/persistent/parquet_column_writer.h"
+#include <parquet/api/writer.h>
+
+namespace kuzu {
+namespace processor {
+
+// ParquetWriter performs the following:
+// openFile: opens the file and create a arrow::io::FileOutputStream object
+// init:
+//   - generate the schema
+//   - calculate the max definition levels and number of primitive nodes
+//   - initialize parquetColumnWriter
+// writeValues : take a vector of ValueVector and pass to parquetColumnWriter
+class ParquetFileWriter : public CSVParquetWriter {
```

Comment:
I would also add some comments on the definition level and repetition level, by one or two examples perhaps. Or put a link for external reference material here.